### ISSUE TITLE
publicOnly incorrectly triggered for local (nested) functions

### ISSUE THREAD

**Issue by Josmithr** [gajus/eslint-plugin-jsdoc#654]
When the `publicOnly` option is specified, the rule seems to be triggered for local functions *within* public members. E.g. an exported function contains (as an implementation detail) a local function used by its own logic.

## Expected behavior
Local (nested) function should not trigger a linter failure if not documented (since it is not public)

## Actual behavior
Local (nested) function triggers a linter failure when not documented (despite not being public)

## ESLint Config
I was able to reproduce the issue with this minimal configuration:
```javascript
module.exports = {
	plugins: ['jsdoc'],
	rules: {
		'jsdoc/require-jsdoc': [
			'error',
			{
				publicOnly: true,
				enableFixer: false,
				require: {
					ClassDeclaration: true,
					ClassExpression: true,
					ArrowFunctionExpression: true,
					FunctionDeclaration: true,
					FunctionExpression: true,
					MethodDefinition: false,
				},
			},
		],
	},
};
```

## ESLint sample

Simple example of a local function triggering the rule.

```typescript
/**
 * Foo
 */
export function foo(): void {
	function bar(): void {
		console.log('bar');
	}

	console.log('foo');
}
```

eslint version: 7.12.1
eslint-plugin-jsdoc version: 30.7.6
